legend-sdlc-generation-service/src/main/resources/generation/service/ServiceExecutionClassGenerator.ftl

```diff
@@ -0,0 +1,56 @@
+<#macro executeMethod withStreamProvider>
+    public Result execute(${executionParameters?map(param-> param.javaSignature)?join(", ")} <#if withStreamProvider><#if executionParameters?size != 0 >,</#if>StreamProvider ${streamProviderParameterName}</#if>)
+    {
+      ExecutionBuilder builder = newExecutionBuilder(${executionParameters?size});
+
+      <#if withStreamProvider>
+        builder = builder.withStreamProvider(${streamProviderParameterName});
+      </#if>
+
+      <#list executionParameters as param>
+          <#if !param.javaMethodType.primitive>
+              if (${param.javaParamName} != null)
+              {
+                  builder = builder.withParameter("${param.legendParamName}", ${param.javaParamName});
+              }
+          <#else>
+              builder = builder.withParameter("${param.legendParamName}", ${param.javaParamName});
+          </#if>
+
+      </#list>
+
+      return builder.execute();
+    }
+</#macro>
```

Comment:
Is this going to produce Java code with proper indentation? It's important that the generated Java code look neat and readable.

Reply:
Few fixes around this:
- Indentation
- Added the copyright (is this required?)
- Ensured line endings are unix (\n)

To assert this behavior, and since the freemarker template is technically a program, I'm asserting the generation output.  This should also aide on visualizing the generated code.